### PR TITLE
libosmium: 2.19.0 → 2.20.0, osmium-tool: 1.15.0 → 1.16.0

### PR DESCRIPTION
**Pull request by sikmir**
## Description of changes
* libosmium: [Changelog](https://github.com/osmcode/libosmium/releases/tag/v2.20.0)
* osmium-tool: [Changelog](https://github.com/osmcode/osmium-tool/releases/tag/v1.16.0)
* pyosmium: fix build (https://hydra.nixos.org/build/235359331)

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 256339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.pyosmium</li>
    <li>python310Packages.pyosmium.dist</li>
    <li>python311Packages.pyosmium</li>
    <li>python311Packages.pyosmium.dist</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libosmium</li>
    <li>osm2pgsql</li>
    <li>osmium-tool</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 256339` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.pyosmium</li>
    <li>python310Packages.pyosmium.dist</li>
    <li>python311Packages.pyosmium</li>
    <li>python311Packages.pyosmium.dist</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libosmium</li>
    <li>osm2pgsql</li>
    <li>osmium-tool</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 256339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>libosmium</li>
    <li>osm2pgsql</li>
    <li>osmium-tool</li>
    <li>python310Packages.pyosmium</li>
    <li>python310Packages.pyosmium.dist</li>
    <li>python311Packages.pyosmium</li>
    <li>python311Packages.pyosmium.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
In a pure shell:

```shell
$ osm2pgsql-replication
/usr/bin/env: 'python3': No such file or directory
```

Even with python in path it fails to find libraries

**Comment by sikmir** [comment]
> In a pure shell:
> 
> ```shell
> $ osm2pgsql-replication
> /usr/bin/env: 'python3': No such file or directory
> ```
> 
> Even with python in path it fails to find libraries

Well, but that's about `osm2pgsql`, https://github.com/NixOS/nixpkgs/pull/256375 is more proper PR to fix it.

File: pkgs/development/python-modules/pyosmium/default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { lib
 , buildPythonPackage
 , fetchFromGitHub
+, fetchpatch
 , cmake
 , python
```
Comment (pbsds) [inline pkgs/development/python-modules/pyosmium/default.nix:6]:
```suggestion
```

**Review by pbsds — APPROVED** [review]
LGTM, but please apply my nit below

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Comment by das-g** [comment]
Result of `nixpkgs-review pr 256339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>libosmium</li>
    <li>osm2pgsql</li>
    <li>osmium-tool</li>
    <li>python310Packages.pyosmium</li>
    <li>python310Packages.pyosmium.dist</li>
    <li>python311Packages.pyosmium</li>
    <li>python311Packages.pyosmium.dist</li>
  </ul>
</details>

**Review by das-g — APPROVED** [review]

**Pull request merged**